py/server/deephaven/table_listener.py

```diff
@@ -188,13 +188,25 @@ def modified_columns(self) -> List[str]:
 
 
 class TableListener(ABC):
-    """An abstract table listener class that should be subclassed by any user table listener class."""
+    """An abstract table listener class that should be subclassed by any user table listener class. It provides a
+    default implementation for the on_error method that simply prints out the error."""
 
     @abstractmethod
     def on_update(self, update: TableUpdate, is_replay: bool) -> None:
         """The required method on a listener object that receives table updates."""
         ...
 
+    def on_error(self, e: Exception) -> None:
+        """The callback method on a listener object that handles the received error. The default implementation simply prints the error.
+
+        Args:
+            e (Exception): the exception that occurred during the listener's execution.
+        """
+        print(f"An error occurred during listener execution: {self}, {e}")
+
+
+def _default_on_error(e: Exception) -> None:
```

Comment:
I would move this above all of the python listener classes, especially if it gets used by them.